File: backend/seed.py
from database import SessionLocal, engine
import models
import auth

print("Creating database bounds...")
# Drop and recreate ensuring absolute clean seeding
models.Base.metadata.drop_all(bind=engine)
models.Base.metadata.create_all(bind=engine)

def seed_data():
    db = SessionLocal()
    print("Initiating Default Seeds...")

    # Users
    admin = models.User(email="admin@hemoi.com", full_name="Super Admin", role=models.RoleEnum.ADMIN, hashed_password=auth.get_password_hash("SecurePassword123!"))
    hospital_user = models.User(email="dispatch@svtc.org", full_name="Valley Trauma", role=models.RoleEnum.HOSPITAL, hashed_password=auth.get_password_hash("HospitalAccess123!"))
    patient_user = models.User(email="john.doe@mail.com", full_name="John Doe", role=models.RoleEnum.DONOR, hashed_password=auth.get_password_hash("PatientPassword123!"))
    
    db.add_all([admin, hospital_user, patient_user])
    db.commit()
    
    # Hospital Tracking
    hospital = models.Hospital(user_id=hospital_user.id, name="Silicon Valley Trauma Center", license_no="SVTC-999-XYZ", location_lat=37.3382, location_lng=-121.8863)
    db.add(hospital)
    db.commit()
    
    # Patient Node
    patient = models.Patient(hospital_id=hospital.id, name="John Doe", blood_group=models.BloodGroupEnum.O_NEG, chronic_conditions="None")
    db.add(patient)
    db.commit()

    # Regional Blood Bank
    regional_bank = models.BloodBank(name="Regional Distro Hub", location_lat=37.3600, location_lng=-121.9400)
    db.add(regional_bank)
    db.commit()

    # Inventory
    for b_type in [models.BloodGroupEnum.O_NEG, models.BloodGroupEnum.O_POS, models.BloodGroupEnum.A_POS]:
         inv = models.Inventory(blood_bank_id=regional_bank.id, blood_group=b_type, units=50)
         db.add(inv)
         
    db.commit()
    print("SUCCESS: Database fully seeded with Production Default Variables.")
    db.close()

if __name__ == "__main__":
    seed_data()

File: backend/prisma/seed.ts
import { PrismaClient, Role, BloodGroup } from '@prisma/client';
import bcrypt from 'bcryptjs';
import { encryptField } from '../src/utils/encryption';

const prisma = new PrismaClient();

async function main() {
  console.log('Initiating database seeding...');

  // Clean existing tables safely
  await prisma.auditLog.deleteMany();
  await prisma.systemLog.deleteMany();
  await prisma.notification.deleteMany();
  await prisma.priorityCase.deleteMany();
  await prisma.inventory.deleteMany();
  await prisma.bloodRequest.deleteMany();
  await prisma.extractedParameter.deleteMany();
  await prisma.aIResult.deleteMany();
  await prisma.medicalReport.deleteMany();
  await prisma.patient.deleteMany();
  await prisma.bloodBank.deleteMany();
  await prisma.hospital.deleteMany();
  await prisma.donor.deleteMany();
  await prisma.user.deleteMany();

  console.log('Cleaned database tables successfully.');

  const adminHash = await bcrypt.hash('SecurePassword123!', 10);
  const hospitalHash = await bcrypt.hash('HospitalAccess123!', 10);
  const bankHash = await bcrypt.hash('BankAccess123!', 10);
  const patientHash = await bcrypt.hash('PatientPassword123!', 10);

  const adminUser = await prisma.user.create({
    data: {
      email: 'admin@raktava.in',
      fullName: 'Aarav Sharma',
      role: Role.ADMIN,
      passwordHash: adminHash,
    },
  });

  const hospitalUser = await prisma.user.create({
    data: {
      email: 'dispatch@apollo.in',
      fullName: 'Dr. Aditya Patel',
      role: Role.HOSPITAL,
      passwordHash: hospitalHash,
    },
  });

  const stJudeUser = await prisma.user.create({
    data: {
      email: 'aiims@raktava.in',
      fullName: 'Dr. Priya Nair',
      role: Role.HOSPITAL,
      passwordHash: hospitalHash,
    },
  });

  const bankUser = await prisma.user.create({
    data: {
      email: 'bank@raktava.in',
      fullName: 'Rajesh Kumar',
      role: Role.BLOOD_BANK,
      passwordHash: bankHash,
    },
  });

  const patientUser = await prisma.user.create({
    data: {
      email: 'amit.verma@mail.in',
      fullName: 'Amit Verma',
      role: Role.PATIENT,
      passwordHash: patientHash,
    },
  });

  const janeUser = await prisma.user.create({
    data: {
      email: 'deepika.sen@mail.in',
      fullName: 'Deepika Sen',
      role: Role.PATIENT,
      passwordHash: patientHash,
    },
  });

  console.log('Seeded Users.');

  const hospital = await prisma.hospital.create({
    data: {
      userId: hospitalUser.id,
      name: 'Apollo Emergency Hospital, Mumbai',
      licenseNo: 'APOLLO-MUM-400001',
      locationLat: 19.0760,
      locationLng: 72.8777,
    },
  });

  const stJudeHospital = await prisma.hospital.create({
    data: {
      userId: stJudeUser.id,
      name: 'AIIMS Emergency Trauma, Delhi',
      licenseNo: 'AIIMS-DEL-110001',
      locationLat: 28.6139,
      locationLng: 77.2090,
    },
  });

  console.log('Seeded Hospitals.');

  const patient = await prisma.patient.create({
    data: {
      hospitalId: hospital.id,
      name: encryptField('Amit Verma'),
      age: 45,
      gender: 'Male',
      bloodGroup: BloodGroup.O_NEG,
      chronicConditions: encryptField('None'),
    },
  });

  const janePatient = await prisma.patient.create({
    data: {
      hospitalId: stJudeHospital.id,
      name: encryptField('Deepika Sen'),
      age: 28,
      gender: 'Female',
      bloodGroup: BloodGroup.A_POS,
      chronicConditions: encryptField('Anemia'),
    },
  });

  console.log('Seeded Patients.');

  const regionalBank = await prisma.bloodBank.create({
    data: {
      userId: bankUser.id,
      name: 'Red Cross Distro Hub, Bengaluru',
      locationLat: 12.9716,
      locationLng: 77.5946,
    },
  });

  console.log('Seeded Blood Bank.');

  const expiryDate = new Date();
  expiryDate.setDate(expiryDate.getDate() + 30);

  const inventoryGroups = [BloodGroup.O_NEG, BloodGroup.O_POS, BloodGroup.A_POS];

  for (const group of inventoryGroups) {
    await prisma.inventory.create({
      data: {
        bloodBankId: regionalBank.id,
        bloodGroup: group,
        units: 50,
        expiryDate: expiryDate,
      },
    });
  }

  console.log('Seeded Inventory.');
  console.log('Database successfully seeded with Production default parameters.');
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
